### PR TITLE
Adding python RAG samples, update existing to use decorators

### PR DESCRIPTION
**Pull request by hallvictoria**
- Added sample for RAG CosmosDB
- Added sample for RAG Kusto
- Converted generic bindings in assistant, chat, and text completion samples to use new decorators (available in azure-functions>=1.20.0b2)
- Updated host.json files to pull from latest extension bundle instead of building it (except for RAG cosmos and kusto samples as using extensions.csproj is necessary)

### CODE REVIEW

File: samples/chat/README.md [hunk 1]
```diff
@@ -24,7 +24,7 @@ Please refer to the root level [README](../../README.md#requirements) for prereq
     | .NET oo-proc | `cd samples/chat/csharp-ooproc && func start` |
     | Node.js | `cd samples/chat/nodejs && npm install && dotnet build --output bin && npm run build && npm run start` |
     | PowerShell | `cd samples/chat/powershell && dotnet build --output bin && func start` |
-    | Python | `cd samples/chat/python && dotnet build --output bin && func start` |
```
Comment (manvkaur) [inline samples/chat/README.md:27]: missed deletion of extensions.csproj?

File: samples/rag-cosmosdb/python/host.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (manvkaur) [inline samples/rag-cosmosdb/python/host.json:1]: missing formatting - use Alt+ Shft + F in VS Code for formatting

File: samples/rag-cosmosdb/python/extensions.csproj [hunk 3]
```diff
@@ -11,7 +11,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <!-- Locally build and reference the extension -->
+    <ProjectReference Include="../../../src/WebJobs.Extensions.OpenAI.CosmosDBSearch/WebJobs.Extensions.OpenAI.CosmosDBSearch.csproj" />
```
Comment (manvkaur) [inline samples/rag-cosmosdb/python/extensions.csproj:14]: are you not able to use just the bundles here?

File: samples/chat/README.md (hunk 1 above)
Comment (hallvictoria) [reply samples/chat/README.md:27]: It is deleted. Not sure why it doesn't show up clearly in the files changed section, but you can see it's not present in the files in this branch

File: samples/rag-cosmosdb/python/extensions.csproj (hunk 3 above)
Comment (hallvictoria) [reply samples/rag-cosmosdb/python/extensions.csproj:14]: Just using the bundles threw this error: error NU1102: Unable to find package Microsoft.Azure.WebJobs.Extensions.OpenAI with version (>= 0.16.0-alpha)

File: samples/chat/README.md (hunk 1 above)
Comment (manvkaur) [reply samples/chat/README.md:27]: I see, its showing as moved to Kusto

**Review by manvkaur — APPROVED** [review]
Looks good, thank you :)

**Pull request merged**